core/pom.xml

```diff
@@ -11,6 +11,11 @@
   <properties>
     <guava.version>20.0</guava.version>
     <gwt.version>2.8.0</gwt.version>
+    <junit.version>4.12</junit.version>
+    <jsr305.version>3.0.1</jsr305.version>
+    <auto-value.version>1.4.1</auto-value.version>
+    <compile-testing.version>0.10</compile-testing.version>
```

Comment:
@cgruber wrote in the review of that code:

> Might not be worth it, but it can prevent a future java version from permitting this to compile unexpectedly.  Maybe too low a risk to be worth the test.

We could also drop our guarantee to test on jdk8. We can still produce java7 classfiles but test them on a higher level (similar to what autovalue and others are doing)